virttest/utils_time.py

```diff
@@ -0,0 +1,244 @@
+        session.cmd(set_timezone_cmd % verify_result)
+        vm_params = vm.params
+        error_context.context("Reboot guest...", logging.info)
+        vm.destroy()
+        vm.create(params=vm_params)
+        vm.verify_alive()
+        session = vm.wait_for_login()
+        if verify_timezone_win(session) is not True:
+            raise exceptions.TestError("Fail to sync guest's timezone.")
+    session.close()
+    return True
+
+
+def execute(cmd, timeout=360, session=None):
+    """
+    Execute command in guest or host, if session is not None return
+    command output in guest else return command ouput in host;
+
+    :param cmd: shell commands;
+    :param session: ShellSession or None;
+    :return: command output string
+    :rtype: str
+    """
+    if session:
+        ret = session.cmd_output(cmd, timeout=timeout)
+    else:
+        ret = process.system_output(cmd)
+    target = session and "guest" or "host"
+    logging.debug("(%s) Execute command('%s')" % (target, cmd))
+    return ret
+
+
+def verify_clocksource(expected, session=None):
+    """
+    Verify if host/guest use the expected clocksource
+    :param expected: expected clocksource
+    :param session: VM session
+    """
+    error_context.context("Check the current clocksource in guest", logging.info)
+    cmd = "cat /sys/devices/system/clocksource/"
+    cmd += "clocksource0/current_clocksource"
+    if expected not in execute(cmd, session):
+        return False
+    return True
+
+
+def sync_time_with_ntp(session=None):
+    """
+    Sync guest or host time with ntp server.
+    :param session: VM session or None.
+    """
+    error_context.context("Sync time from ntp server", logging.info)
+    cmd = "ntpdate clock.redhat.com; hwclock -w"
+    return execute(cmd, session)
+
+
+def update_clksrc(vm, image_filename, grub_file, kernel_cfg_pattern, clksrc=None):
+    """
+    Update linux guest's clocksource and re-boot guest.
+
+    :params: vm: Virtual machine for vm.
+    :params: image_filename: Current image's file path.
+    :params: grub_file: VM's grub file path.
+    :params: kernel_cfg_pattern: Pattern to get current kernel line.
+    :params: clksrc: Expected clocksource.
+    """
+    if vm.is_alive():
+        error_context.context("Shutdown guest", logging.info)
+        vm.destroy()
+    update_clksrc_with_kernel_line(image_filename, grub_file,
+                                   kernel_cfg_pattern, clksrc)
+    error_context.context("Boot the guest", logging.info)
+    vm.create(vm.params)
+    vm.verify_alive()
+
+
+def update_clksrc_with_kernel_line(image_filename, grub_file,
+                                   kernel_cfg_pattern, clksrc=None):
+    """
+    Update clocksource in kernel line with libguestfs.
+
+    :params: vm: Virtual machine for vm
+    :params: image_filename: Current image's file path.
+    :params: grub_file: VM's grub file path.
+    :params: kernel_cfg_pattern: Pattern to get current kernel line.
```

Comment:
`%s/:params: kernel_cfg_pattern:/:params kernel_cfg_pattern:` please fix the same in elsewhere.